Invalid none key

joverlee521 opened the pull request:
### Description of proposed changes
- Updates the export v2 schema to make "none" an invalid key for colorings and branch labels. 
- Updates `export v2` to warn about invalid "none" key and ignore it in exports. 
- Does _not_ include changes in `export v2` for branch labels to ignore the "none" key because it is currently hard-coded to only export "clades" and "aa" as branch labels. We should ignore "none" keys for branch labels once we allow for arbitrary labels in #728.


### Related issue(s)
<!-- Start typing the name of a related issue and GitHub will auto-suggest the issue number for you.  -->
Related to https://github.com/nextstrain/auspice/pull/1618

### Testing
* Added new functional test for export v2 using a "none" coloring key.

<!-- 🙌 Thank you for contributing to Nextstrain! ✨ -->

### Checklist

- [x] Add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR that are end user focused. Keep headers and formatting consistent with the rest of the file.

tsibley commented on augur/data/schema-export-v2.json:
```diff
@@ -169,7 +169,8 @@
                                 "description": "They key used to access the value of this coloring on each node",
-                                "type": "string"
+                                "type": "string",
+                                "not": {"enum": ["none"]}
```
```suggestion
                               "not": {"const": "none"}
```

is another way to spell this, but I guess we might want to disallow other specific values in the future…

joverlee521 replied:
Updated to `const` in 7736ca3ba4486ee6a917aa6000a6551a7f29bd82

tsibley commented:
Looks good, needs a changelog entry though.

joverlee521 commented:
Rebased and added changelog entry. I've also made additional changes to make "none" an invalid `--metadata-columns` value since they are added directly as node_attrs that can clash with Auspice's use of "none" to hide tip labels. 

This is ready for a fresh review after ~2 years 😅

victorlin commented on CHANGES.md:
```diff
@@ -5,6 +5,8 @@
 ### Features
 
 * ancestral: Add `--seed` argument to enable deterministic inference of root states by TreeTime. [#1690][] (@huddlej)
+* export v2: The string "none" is now an invalid value for `--color-by-metadata` and `--metadata-columns` options to prevent clashes with Auspice's internal use of "none". [#1113][] (@joverlee521)
+* schema: The string "none" is now an invalid branch label, node_attr key, and coloring key. [#1113][] (@joverlee521)
```
Are these technically https://github.com/nextstrain/augur/labels/breaking changes to be put in the "Major Changes" section?

joverlee521 replied:
Ah yes! Thanks for catching that. Will rebase and update changelog after latest release is complete.

joverlee521 commented:
Did one final rebase to ensure the changelog isn't messed up after the latest release. Will merge once the tests pass.
